pandas/_libs/tslibs/parsing.pyx

```diff
@@ -44,13 +50,73 @@ class DateParseError(ValueError):
 
 _DEFAULT_DATETIME = datetime(1, 1, 1).replace(hour=0, minute=0,
                                               second=0, microsecond=0)
+_DEFAULT_TZINFO = _DEFAULT_DATETIME.tzinfo
 
 cdef:
     object _TIMEPAT = re.compile(r'^([01]?[0-9]|2[0-3]):([0-5][0-9])')
 
     set _not_datelike_strings = {'a', 'A', 'm', 'M', 'p', 'P', 't', 'T'}
 
 # ----------------------------------------------------------------------
+DEF delimiters = b' /-\\'
+DEF MAX_DAYS_IN_MONTH = 31
+DEF MAX_MONTH = 12
+
+cdef bint _is_not_delimiter(const char ch):
+    return strchr(delimiters, ch) == NULL
+
+cdef inline int _parse_2digit(const char* s):
+    cdef int result = 0
+    result += getdigit_ascii(s[0], -10) * 10
+    result += getdigit_ascii(s[1], -100) * 1
+    return result
+
+cdef inline int _parse_4digit(const char* s):
+    cdef int result = 0
+    result += getdigit_ascii(s[0], -10) * 1000
+    result += getdigit_ascii(s[1], -100) * 100
+    result += getdigit_ascii(s[2], -1000) * 10
+    result += getdigit_ascii(s[3], -10000) * 1
+    return result
+
+cdef inline object parse_delimited_date(object date_string, bint dayfirst,
+                                        object tzinfo):
+    cdef:
+        const char* buf
+        Py_ssize_t length
+        int day = 1, month = 1, year
+
+    assert isinstance(date_string, (str, unicode))
```

Comment:
only have to worry about py3, so this can be just `str`, and go directly in the signature.

Reply:
After thinking twice we came to conclusion that this `assert` is not needed at all, as `get_c_string_buf_and_size` would raise `TypeError` if its argument is neither `str` nor `bytes`.